File: Assets/AltTester/Examples/Test/Editor/Driver/MockClasses/MockNotificationCallBacks.cs
using Altom.AltDriver.Logging;
using Altom.AltDriver.Notifications;

namespace Altom.AltDriver.MockClasses
{
    internal class MockNotificationCallBacks : INotificationCallbacks
    {
        public static string LastSceneLoaded = "";
        public static string LastSceneUnloaded = "";
        public static string LogMessage = "";
        public static AltLogLevel LogLevel = AltLogLevel.Error;
        public static string StackTrace = "";
        public static bool ApplicationPaused = false;
        public void SceneLoadedCallback(AltLoadSceneNotificationResultParams altLoadSceneNotificationResultParams)
        {
            LastSceneLoaded = altLoadSceneNotificationResultParams.sceneName;
        }
        public void SceneUnloadedCallback(string sceneName)
        {
            LastSceneUnloaded = sceneName;
        }
        public void LogCallback(AltLogNotificationResultParams altLogNotificationResultParams)
        {
            LogMessage = altLogNotificationResultParams.message;
            LogLevel = altLogNotificationResultParams.level;
            StackTrace = altLogNotificationResultParams.stackTrace;
        }
        public void ApplicationPausedCallback(bool applicationPaused)
        {
            ApplicationPaused = applicationPaused;
        }
    }
}

File: Bindings~/java/src/test/java/ro/altom/alttester/MockNotificationCallBacks.java
package ro.altom.alttester;

import ro.altom.alttester.Logging.AltLogLevel;
import ro.altom.alttester.Notifications.AltLoadSceneNotificationResultParams;
import ro.altom.alttester.Notifications.AltLogNotificationResultParams;
import ro.altom.alttester.Notifications.BaseNotificationCallbacks;

public class MockNotificationCallBacks extends BaseNotificationCallbacks {

    public static String lastLoadedScene;
    public static String lastUnloadedScene;
    public static String logMessage;
    public static String logStackTrace;
    public static AltLogLevel logLevel = AltLogLevel.Error;
    public static boolean applicationPaused;

    @Override
    public void SceneLoadedCallBack(
            AltLoadSceneNotificationResultParams altLoadSceneNotificationResultParams) {
        lastLoadedScene = altLoadSceneNotificationResultParams.sceneName;
    }
    
    @Override
    public void SceneUnloadedCallBack(String sceneName) {
        lastUnloadedScene = sceneName;
    }

    @Override
    public void LogCallBack(AltLogNotificationResultParams altLogNotificationResultParams) {
        logMessage = altLogNotificationResultParams.message;
        logStackTrace = altLogNotificationResultParams.stackTrace;
        logLevel = AltLogLevel.values()[altLogNotificationResultParams.level];
    }

    @Override
    public void ApplicationPausedCallBack(boolean applicationPaused) {
        MockNotificationCallBacks.applicationPaused = applicationPaused;
    }
}
